source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -88,6 +135,28 @@ bool Utility::dnsNameMatch(absl::string_view dns_name, absl::string_view pattern
     }
   }
 
+  // https://www.rfc-editor.org/rfc/rfc6125#section-6.4.3 part #3
+  // Match a presented identifier in which the wildcard character is not the only
+  // character of the label and don't match patter if the wildcard character is
+  // embedded within an A-label (A-label always starts with the ACE prefix "xn--")
+  // (e.g., baz*.example.net and *baz.example.net and b*z.example.net would
+  //       be taken to match baz1.example.net and foobaz.example.net and
+  //       buzz.example.net, respectively)
+  size_t pattern_left_label_len = lower_case_pattern.find('.');
+  size_t dns_name_left_label_len = lower_case_dns_name.find('.');
+  // Only the left-most label in the pattern contains wildcard '*' and is not an A-label
+  if ((pattern_left_label_len != std::string::npos) &&
```

Comment:
This conditional is too complicated. Please break it up, using some temporaries, and possibly some early returns. An `absl::string_view` of the first component may also help in readability. Consider using `absl::StrSplit()` to get the first and remaining components as string_view.

Reply:
Got it. Let me simplify this for better readability. Thanks for the suggestion.

Reply:
Broke it down. Now code looks much simpler after using `absl::StrSplit`.